### ISSUE TITLE
Triggers in saved games saved through Action 500 does not work properly

### ISSUE THREAD

**Issue by ghost** [Phobos-developers/Phobos#863]
### First check

- [X] The issue happens on the **latest official** version of Phobos and wasn't fixed yet.
- [X] I agree to elaborate the details if requested and provide thorough testing if the bugfix is implemented.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search and read the issue list to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, gifs, videos, debug and exception logs, etc)*.

### Description

**Triggers** enabled by _trigger_ with `Action500` won't work in the saved game if **Triggers**' Event is `13 Elapsed time` with para `0`.

Please refer to the following text `Expected behaviour` for details

### Phobos Version

Release Build 0.3.0.0

### Conditions to reproduce

Mental Omega 3.3.6 with Syringe v0.7.3.0

### INI code

```ini
[Actions]
01000000=3,500,4,0,0,0,0,0,A,53,2,01000002,0,0,0,0,A,14,0,3,0,0,0,0,A
01000002=1,14,0,3,0,0,0,0,A

[Events]
01000000=1,13,0,4
01000002=1,13,0,0

[Tags]
01000001=0,1 Save Game 1,01000000
01000003=0,2 Change House 1,01000002

[Triggers]
01000000=UnitedStates,<none>,1 Save Game,0,1,1,1,0
01000002=UnitedStates,<none>,2 Change House,1,1,1,1,0

[Units]
0=UnitedStates House,HTNK,256,81,83,64,Guard,01000003,0,-1,0,-1,0,1
1=UnitedStates House,HTNK,256,81,85,64,Guard,01000001,0,-1,0,-1,0,1
```

### Steps to reproduce

1. Add the Triggers in `INI code` to a new campaign map
2. If the mod is not Mental Omega, change the **_United States House_** to any House on the map except for No.3 one
3. If the mod does not include unit **_HTNK_**, please change it
4. Start the game and wait
5. Load the Saved Game by `Action500`

### Expected behaviour

**Trigger A** contains three actions, the first is `Action500` to save the game, the second one is `Action53` to enable **Trigger B** activate, the third one `Action14` changing **Unit A** from one house to another.
**Trigger B** changes **Unit B** from one house to another.
**Trigger B** is disabled, requires **Trigger A** to enable it (by `Action53`)
Event of **Trigger B** is `13 Elapsed time` with para `0`.

> `[Actions]`
`01000000=3,500,4,0,0,0,0,0,A,53,2,01000002,0,0,0,0,A,14,0,3,0,0,0,0,A`
`01000002=1,14,0,3,0,0,0,0,A`
`[Units]`
`0=UnitedStates House,HTNK,256,81,83,64,Guard,01000003,0,-1,0,-1,0,1`
`1=UnitedStates House,HTNK,256,81,85,64,Guard,01000001,0,-1,0,-1,0,1`

Under expected situation, **Trigger A** changes **Unit A** to another house (Red colour) , and then enables **Trigger B**, which changes **Unit B** to another house (Red colour).
If I load the saved game, they should be Red colour.

### Actual behaviour

At first, it worked normally: **Unit A** was changed to House No3 (Red colour), and **Unit B** was changed to House No3, the game was saved.
![image](https://user-images.githubusercontent.com/79000292/232322450-1ea73329-fd9a-42e3-9945-85a48df6f930.png)
**BUT** I loaded the saved game (which is saved by `Action 500`), none of them changes its house (Blue colour) !
![image](https://user-images.githubusercontent.com/79000292/232322485-eff66374-d55c-4025-bafa-46199499c0b5.png)

Then I tried to exchange the order of `Action500` with others. 

From
> `[Actions]`
`01000000=3,500,4,0,0,0,0,0,A,53,2,01000002,0,0,0,0,A,14,0,3,0,0,0,0,A`
`01000002=1,14,0,3,0,0,0,0,A`

To
> `[Actions]`
`01000000=3,53,2,01000002,0,0,0,0,A,500,4,0,0,0,0,0,A,14,0,3,0,0,0,0,A`
`01000002=1,14,0,3,0,0,0,0,A`

In the saved game, only **Unit B** which attached with `Tag01000003` changed to the new house.
`Tag01000003` is for `Trigger01000002`
![image](https://user-images.githubusercontent.com/79000292/232321946-20cfa544-8abf-4a63-ba31-1ca0baddc0ff.png)

### Additional context

Example Map for Mental Omega
Please change the suffix name `.txt` to `.map` (`.map` files are not allowed for upload)
[areddawn.txt](https://github.com/Phobos-developers/Phobos/files/11242606/areddawn.txt)

**Comment by chaserli** [comment]
For all non-standard game saves (t event or hotkey), I think we should just send a signal to notify the game should be saved, and execute saving game at a safer place.

**Comment by chaserli** [comment]
probably fixed in 83d3b2a
